### PR TITLE
New example: MRSA community

### PR DESCRIPTION
**Pull request by gvegayon**
This pull request includes several changes to the `epiworld` project, focusing on template parameter defaults, error message clarity, and the addition of a new community-hospital model example. The most important changes are summarized below:

### Template Parameter Defaults:
* Updated various template definitions to use `EPI_DEFAULT_TSEQ` as the default type for `TSeq` instead of `int`. This change improves code consistency and flexibility. (`epiworld.hpp` [[1]](diffhunk://#diff-3bdaf2103de7e4092beaf2f4c5e966635c155d3bd36ca531765a5337db0bd5faL598-R598) [[2]](diffhunk://#diff-3bdaf2103de7e4092beaf2f4c5e966635c155d3bd36ca531765a5337db0bd5faL13081-R13081) [[3]](diffhunk://#diff-3bdaf2103de7e4092beaf2f4c5e966635c155d3bd36ca531765a5337db0bd5faL13240-R13240) [[4]](diffhunk://#diff-3bdaf2103de7e4092beaf2f4c5e966635c155d3bd36ca531765a5337db0bd5faL13408-R13408); `include/epiworld/agent-meat-virus-sampling.hpp` [[5]](diffhunk://#diff-f4b7fd099bbb7906acc8f83ec13a8b030108cf35cb95a25222772db02a3dd22bL23-R23) [[6]](diffhunk://#diff-f4b7fd099bbb7906acc8f83ec13a8b030108cf35cb95a25222772db02a3dd22bL182-R182) [[7]](diffhunk://#diff-f4b7fd099bbb7906acc8f83ec13a8b030108cf35cb95a25222772db02a3dd22bL350-R350); `include/epiworld/misc.hpp` [[8]](diffhunk://#diff-1b7c8676e5cb3b6335e78e62fd709b9fc6124ed179e0a6878156d72368316c03L127-R127)

### Error Message Clarity:
* Improved the clarity of an error message in the `Model<TSeq>::operator()` method by adding single quotes around the parameter name. (`epiworld.hpp` [[1]](diffhunk://#diff-3bdaf2103de7e4092beaf2f4c5e966635c155d3bd36ca531765a5337db0bd5faL7664-R7664); `include/epiworld/model-meat.hpp` [[2]](diffhunk://#diff-501cd8978f05e81715765fdf2df0552b9f9fbaf8d22a9f4051444bd201b690b6L717-R717)

### Community-Hospital Model Example:
* Added a new example model that simulates the interaction between community and hospital populations. This includes updates to the `Makefile`, a detailed `README.md`, and a new `main.cpp` file with the model implementation. (`examples/14-community-hosp/Makefile` [[1]](diffhunk://#diff-ba5b789607fa0aa094014148e9757f611227a8d71269b80e515fa02455d8049bR1-R2); `examples/14-community-hosp/README.md` [[2]](diffhunk://#diff-c4fea3e065646273be9c3c13713fa4fd13474b00154a6a04b10336f7a4a60686R1-R99); `examples/14-community-hosp/main.cpp` [[3]](diffhunk://#diff-6151647f985a9112c5d8ca4faae4a7617882e9bee7d3c470e2fa5119880568f6R1-R133)
* Updated the main `Makefile` to include the new community-hospital example. (`examples/Makefile` [examples/MakefileL11-R12](diffhunk://#diff-ae8bb09b4bbb919309d4c8bb8ffc46e031bf63b7274502171a183a4dc1138f53L11-R12))

### CODE REVIEW

File: examples/14-community-hosp/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+- Prob hospitalization: 0.1
+- Prob recovery: 0.33
+- Discharge infected: 0.1
+
+Here is an example of the run
+
+```bash
+root ➜ /workspaces/epiworld/examples/14-community-hosp (example-karim) $ make
+g++ -std=c++17 -O3 -g main.cpp -o main.o
+root ➜ /workspaces/epiworld/examples/14-community-hosp (example-karim) $ ./main.o
```
Comment (apulsipher) [inline examples/14-community-hosp/README.md:22]: I'd suggest updating this example to remove `(example-karim)` for additional clarity

File: examples/14-community-hosp/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ - Susceptible              0.98  0.02  0.00
+ - Infected                 0.32  0.61  0.07
+ - Infected (hospitalized)  0.35  0.07  0.58
+```
+
+## Notes
+
+An few key observations from this example.
+
+1. **We have a sampling function that exclude population**. These two functions are used in the update functions so, when susceptible (in the community) sample, the sampling excludes individuals who are hospitalized. Likewise, hospitalized
```
Comment (apulsipher) [inline examples/14-community-hosp/README.md:70]: This last sentence appears to be incomplete, or at least I'm not following it. "Likewise, hospitalized..." what?

File: examples/14-community-hosp/main.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
+#include "../../epiworld.hpp"
+
+/***
+ * A concrete example would be a model that includes two populations.
```
Comment (apulsipher) [inline examples/14-community-hosp/main.cpp:4]: Should be reworded to "A concrete example with a model that includes two populations:"

File: examples/14-community-hosp/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
+# Community-hospital model
+
+In this model, we have three states:
+
+- Susceptible individuals (in the community),
+- Infected individuals (in the community), and
+- Infected hospitalized individuals.
+
+Susceptible individuals may become hospitalized or not (so they have two possible transitions), and infected individuals may recover or (if hospitalized) stay infected but be discharged (so they go back as infected but in the community).
```
Comment (apulsipher) [inline examples/14-community-hosp/README.md:9]: I think this needs to clarification that "When Susceptible individuals become Infected, they may also be come hospitalized (so they have two possible transitions)." I didn't understand this part immediately on the first read-through, at it became clearer in the code comments below.

**Review by apulsipher — COMMENTED** [review]
Generally this looks good! There are a few changes to make, and some of the automated checks are failing. Of the checks, the Doxygen check seems to be the biggest concern. Do we need to fix that test or is that something you can fix on this branch?

File: examples/14-community-hosp/README.md (hunk 4 above)
Comment (gvegayon) [reply examples/14-community-hosp/README.md:9]: Good point. I think we need to add a mermaid diagram here. That will make things easier. I'm putting a pin to it so we can address it later.

File: examples/14-community-hosp/main.cpp (hunk 3 above)
Comment (gvegayon) [reply examples/14-community-hosp/main.cpp:4]: I'm addressing this in #34.

File: examples/14-community-hosp/README.md (hunk 2 above)
Comment (gvegayon) [reply examples/14-community-hosp/README.md:70]: I'm addressing this in #34.

**Comment by gvegayon** [comment]
Closing this in favor of https://github.com/UofUEpiBio/epiworld/pull/34.

**Pull request closed without merging**